sdk/src/test/scala/com/horizen/account/wallet/AccountWalletTest.scala

```diff
@@ -195,4 +195,40 @@ class AccountWalletTest
     assertEquals("SidechainWallet different exception expected during removing new Secret.", expectedException, failureResult.failed.get)
   }
 
+  @Test
+  def testGenerateNextSecret(): Unit = {
```

Comment:
1) need to extend the test cases according to the issues identified in the `WalletgenerateNextSecret` method.
2) add regression tests to be sure that we generate the expected secret.

Reply:
Actually, the test covers exactly the case described by NCC as a vulnerability. The case when we have for example a list of secrets, one is removed (not the last one), one is added, and we get secret duplication previously (before changes in this PR). @i-Alex could you tell me please what other cases should I cover? NCC described the case I already covered